libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -426,10 +435,56 @@ void AP_Airspeed::read(uint8_t i)
         state[i].healthy = false;
     }
 
-    state[i].last_update_ms = AP_HAL::millis();
+    state[i].last_update_ms = now;
+}
+
+void AP_Airspeed::update_vel_dot(void)
+{
+    // Update and average speed rate of change
+    // Get DCM
+    const Matrix3f &rotMat = AP_AHRS::get_singleton()->get_rotation_body_to_ned();
+    // Calculate speed rate of change
+    float temp = rotMat.c.x * GRAVITY_MSS + AP::ins().get_accel().x;
+    // take 5 point moving average
+    vel_dot = vdot_filter.apply(temp);
+}
+
+void AP_Airspeed::update_EAS(void)
+{
+    float _EAS = get_airspeed();
+
+    // Calculate time in seconds since last update
+    uint64_t now = AP_HAL::micros64();
+    float DT = (now - _update_speed_last_usec) * 1.0e-6f;
+    _update_speed_last_usec = now;
+
+    // Reset states of time since last update is too large
+    if (DT > 1.0f) {
+        _EAS_state = _EAS;
+        _integETAS_state = 0.0f;
+        DT            = 0.1f; // when first starting TECS, use a
+        // small time constant
+    }
+
+    // Implement a second order complementary filter to obtain a
+    // smoothed airspeed estimate
+    // airspeed estimate is held in _EAS_state
+    constexpr float _spdCompFiltOmega = 2.0f;
+    float aspdErr = _EAS - _EAS_state;
+    float integDTAS_input = aspdErr * _spdCompFiltOmega * _spdCompFiltOmega;
+    // Prevent state from winding up
+    if (_EAS_state < 3.1f) {
+        integDTAS_input = MAX(integDTAS_input , 0.0f);
+    }
+    _integETAS_state = _integETAS_state + integDTAS_input * DT;
+    float EAS_input = _integETAS_state + vel_dot + aspdErr * _spdCompFiltOmega * M_SQRT2;
+    _EAS_state = _EAS_state + EAS_input * DT;
+    // limit the airspeed to a minimum of 3 m/s
+    _EAS_state = MAX(_EAS_state, 3.0f);
 }
 
 // read all airspeed sensors
+// called at 100Hz
```

Comment:
Can't count on this, also you seem to have read the table wrong, both plane and rover are 10Hz.

Reply:
right... wrong column